crates/turborepo-auth/src/auth/mod.rs

```diff
@@ -48,6 +43,15 @@ where
     }
 }
 
+/// Options for logging out.
+pub struct LogoutOptions<'a, T> {
+    pub ui: &'a UI,
+    pub api_client: &'a T,
+
+    /// The path where we should look for the token to logout.
+    pub path: &'a AbsoluteSystemPathBuf,
```

Comment:
For borrowed/owned type pairs like `&[T]/Vec<T>`/`&str/String`/`&Path/PathBuf` you almost always want to use the borrowed variant instead of `&String`. The gist is that for all of these types the owned variant has a pointer to some heap allocated memory. So `&Vec<T>` is a pointer to `Vec` which is on the stack which then contains a pointer to the heap memory. If convert `&Vec<T>` to `&[u8]` you skip needing to resolve that first pointer and following it will take you directly to the heap memory.
```suggestion
    pub path: &'a AbsoluteSystemPath,
```

Reply:
I always get confused which ASP is the borrowed variant vs the owned one. I've thought about if we could give these more straightforward names but haven't come up with anything - might be worth discussing as a team since I know Mehul also mentioned he got confused by these names

Reply:
They're based off of `&Path`/`PathBuf` from the [stdlib](https://doc.rust-lang.org/std/path/index.html).